### ISSUE TITLE
Requested package versions are ignored in some situations

### ISSUE THREAD

**Issue by jacebrowning** [kennethreitz/pipenv#148]
With the latest version (3.2.10), installing one package that depends on another causes the first package's dependencies to get locked:

```
$ pipenv install --dev pytest-django==3.1.2 pytest==2.9.2 --lock
```

In this case `pytest-django` specifies `install_requires=['pytest>=2.9']`. I would expect my request to specifically install `pytest==2.9.2` to be present in the lock file, but it is not:

```
...
    "develop": {
        "pytest-django": {
            "hash": "sha256:00995c...",
            "version": "==3.1.2"
        },
        "pytest": {
            "hash": "sha256:da0ab5...",
            "version": "==3.0.6"
        },
...
```

**Comment by kennethreitz** [comment]
this is a bug

**Comment by kennethreitz** [comment]
@nateprewitt i think this is a result of the work you were thinking of rolling back

**Comment by nateprewitt** [comment]
Hmm, it seems like it very well could be related. I'm betting it's how we're now parsing all downloaded dependencies. If it encounters the dependency requiring >=2.9 first, it will download it. Then when we iterate through the downloads, the 3.0.6 dependency comes last and overrides the previous version.

I'll take a closer look here in a moment.

**Comment by nateprewitt** [comment]
Hmm, @jacebrowning are you certain this has ever worked? I'm testing clear back to 3.0.1 and seeing an identical Pipfile.lock generated each time. This is definitely a bug, I'm just trying to verify that it was preexisting and not directly related to how we're now parsing dependency names.

**Comment by nateprewitt** [comment]
Resolved with #149, thanks for letting us know about this @jacebrowning! It looks like this bug has been around for a while.

**Comment by jacebrowning** [comment]
@nateprewitt I can't confirm that this ever worked. I'll retest with the next release.
